### PR TITLE
fix `json.stringify` escape

### LINKED ISSUE #812: json.stringify does not handle the escaped characters
details #811

### PR DESCRIPTION
**Pull request by yj-qin**
fix #812

### CODE REVIEW

File: json/json.mbt [hunk 1]
```diff
@@ -121,6 +121,25 @@ pub fn stringify(self : JsonValue) -> String {
   buf.to_string()
 }
 
+fn escape(str : String) -> String {
+  let buf = Buffer::new()
```
Comment (qazxcdswe123) [inline json/json.mbt:125]: set `size_hint` to `str.length()` to avoid some reallocation?

File: json/json.mbt [hunk 2]
```diff
@@ -121,6 +121,25 @@ pub fn stringify(self : JsonValue) -> String {
...
+  for i = 0; i < str.length(); i = i + 1 {
+    let c = str[i]
+    match c {
+      '"' | '\\' | '/' => {
+        buf.write_char('\\')
```
Comment (qazxcdswe123) [inline json/json.mbt:130]: Maybe use string interpolation `\\\{c}`

```
'"' | '\\' | '/' => buf.write_string("\\\{c}")
```

File: json/json.mbt [hunk 3]
```diff
@@ -121,6 +121,25 @@ pub fn stringify(self : JsonValue) -> String {
...
+        buf.write_char(c)
+      }
+      '\n' => buf.write_string("\\n")
+      '\r' => buf.write_string("\\r")
+      '\b' => buf.write_string("\\b")
+      '\t' => buf.write_string("\\t")
```
Comment (qazxcdswe123) [inline json/json.mbt:136]: https://stackoverflow.com/questions/19176024/how-to-escape-special-characters-in-building-a-json-string

According to this.
Probably missing `\f` and `\u`?

Comment (yj-qin) [reply json/json.mbt:141]: Thanks!

**Comment by CAIMEOX** [comment]
Still fails at the following case:
```moonbit
let json : @json.JsonValue = { "\x0c": True }
```

**Pull request merged**